flutter/lib/screens/qr_code_screen.dart

```diff
@@ -0,0 +1,138 @@
+
+  late Timer _timer;
+  int _time = 0;
+  String _timeInString = "";
+
+  //Offline handler
+  StreamSubscription? _offlineStreamSubscription;
+
+  @override
+  void initState() {
+    super.initState();
+
+    //Simple connection checker object init
+    SimpleConnectionChecker _simpleConnectionChecker = SimpleConnectionChecker()
+      ..setLookUpAddress('pub.dev'); //Optional method to pass the lookup string
+    _offlineStreamSubscription = _simpleConnectionChecker.onConnectionChange.listen((connected) {
+      if(!connected){
+        //Launch offline screen when the device is offline
+        Navigator.of(context).pushNamedAndRemoveUntil(OfflineScreen.routeName, (route) => false);
+      }
+    });
+
+    //Call the seed API
+    var provider = Provider.of<SuperFormulaProvider>(context, listen: false);
+
+    provider.getSeed(firstTime: true,context: context);
+
+    //Timer to update time
+    _timer = Timer.periodic(const Duration(seconds: 1), (Timer t) async{
+
+      //Get NTP time
+      DateTime mySystemTime = DateTime.now().toUtc();
+
+      //Update the Seed[QR Code] every minute
+      if (mySystemTime.second==0) {
+        if(mounted) provider.getSeed(firstTime: false,context: context);
+      }
+
+      //Update the timer text
+      if (mounted) {
+        setState((){
+          //Update the time,Show time in decreasing order
+          _time = 59 - mySystemTime.second;
+
+          //Update the text
+          _timeInString = (_time).toString();
+        });
+      }
+    });
+  }
+
+  @override
+  void dispose() {
+    _offlineStreamSubscription?.cancel();
+    _timer.cancel();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+
+    //Objects
+
+    var _provider = Provider.of<SuperFormulaProvider>(context);
+    String _seed = _provider.seed;
+
+    DOWNLOAD_STATE _downloadState = _provider.qrCodeDownloadingState;
+
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text("QR Code"),
+      ),
+      body: SizedBox(
+        width: double.infinity,
+        height: double.infinity,
+        child: _downloadState==DOWNLOAD_STATE.DOWNLOADING?
+           Center(
+             child: Lottie.asset('assets/animations/circle.json',width: ScreenUtil().setHeight(600),height: ScreenUtil().setHeight(600)),
+           ) :Padding(
+          padding: EdgeInsets.symmetric(vertical: ScreenUtil().setHeight(80)),
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.start,
+            crossAxisAlignment: CrossAxisAlignment.center,
+            children: [
+              Visibility(
+                visible: _seed.isNotEmpty && _timeInString.isNotEmpty,
+                child: QrImage(
+                  data: _seed,
+                  version: QrVersions.auto,
+                  size: ScreenUtil().setHeight(300),
+                ),
+              ),
+              SizedBox(
+                height: ScreenUtil().setHeight(48),
+              ),
```

Comment:
**question:** since the widget above this one can potentially not be visible, should this be surrounded by logic as well that only displays it if the widget above is visible?

Reply:
Yes, the logic should be applied to the parent column widget as you mentioned.